runbotics/runbotics-desktop/src/action/desktop/desktop.action-handler.ts

```diff
@@ -0,0 +1,292 @@
+    private readonly system: string = process.platform;
+    private readonly mouseConfig: MouseConfig = {
+        autoDelayMs: 50,
+        mouseSpeed: 600,
+    };
+    private readonly keyboardConfig: KeyboardConfig = {
+        autoDelayMs: 50,
+    };
+    // screen config -> configure directory for imageResource
+    
+    constructor() {
+        super();
+        mouse.config = this.mouseConfig;
+        keyboard.config = this.keyboardConfig;
+    }
+
+    async click(input: DesktopClickActionInput): Promise<void> {
+        const button: Button = input.mouseButton === MouseButton.LEFT ? Button.LEFT : Button.RIGHT;
+        
+        if (input.clickTarget === ClickTarget.POINT) {
+            this.checkPoint(input.x, input.y);
+            this.moveMouseToPoint(input.x, input.y);
+        } else {
+            const region = input.region;
+            this.checkRegion(region);
+            await this.moveMouseToRegion(+region.left, +region.top, +region.width, +region.height);
+        }
+        
+        if (input.doubleClick) {
+            await mouse.doubleClick(button);
+        } else {
+            await mouse.click(button);
+        }
+    }
+
+    async type(input: DesktopTypeActionInput): Promise<void> {
+        const optionalKey = input.text.substring(4); 
+        if (input.text.startsWith('Key.') && Object.keys(Key).includes(optionalKey)) {
+            await keyboard.type(Key[optionalKey]);
+        } else {
+            await keyboard.type(input.text);
+        }
+    }
+
+    async copy(input: DesktopCopyActionInput): Promise<void> {
+        if (input?.text) {
+            await clipboard.setContent(input.text);
+        } else {
+            const superKey: Key = this.getSuperKey();
+            await this.performKeyboardShortcut([superKey, Key.C]);
+        } 
+    }
+
+    async paste(): Promise<void> {
+        const content = await clipboard.getContent();
+        await keyboard.type(content);
+    }
+
+    async selectWithCursor(input: DesktopSelectWithCursorActionInput): Promise<void> {
+        this.checkPoint(input.startX, input.startY);
+        this.checkPoint(input.endX, input.endY);
+        await this.moveMouseToPoint(input.startX, input.startY);
+        await mouse.pressButton(Button.LEFT);
+        await this.moveMouseToPoint(input.endX, input.endY);
+        await mouse.releaseButton(Button.LEFT);
+    }
+
+    async readContentFromClipboard(): Promise<DesktopReadContentFromClipboardActionOutput> {
+        const content = await clipboard.getContent();
+        return content;
+    }
+
+    async takeScreenshot(input: DesktopTakeScreenshotActionInput): Promise<DesktopTakeScreenshotActionOutput> {
+        // prevents from taking screenshot too fast
+        await sleep(2000);
+        const region = input?.region;
+        const fileName = input?.fileName ?? uuidv4();
+        const filePath = input?.filePath ?? path.join(process.cwd(), 'temp');
```

Comment:
this temp directory should be in utils file or smthing like that. There is at least one more usage in runtime service. Should be in a single place so we prevent possible bugs in the future